rafthttp/http.go

```diff
@@ -257,18 +257,14 @@ func (h *streamHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	p := h.peerGetter.Get(from)
 	if p == nil {
-		if urls := r.Header.Get("X-Server-Peers"); urls != "" {
-			h.tr.AddPeer(from, strings.Split(urls, ","))
-		}
-		p = h.peerGetter.Get(from)
-	}
-
-	if p == nil {
 		// This may happen in following cases:
 		// 1. user starts a remote peer that belongs to a different cluster
 		// with the same cluster ID.
 		// 2. local etcd falls behind of the cluster, and cannot recognize
 		// the members that joined after its current progress.
+		if urls := r.Header.Get("X-Server-Peers"); urls != "" {
```

Comment:
i feel it is better to call this X-PeerURLs? This was peerid=url array, now it is changed.